src/app/pages/courses/courses.routes.ts

```diff
@@ -1,10 +1,33 @@
 import { Routes } from '@angular/router';
 
 import { AppRoutePath } from '@app/enums/app-route-path.enum';
+import { AuthGuard } from '@app/guards/auth/auth.guard';
+import { getStaticBreadcrumbsStepsByRoutePath } from '@app/util/util';
 
-import { CourseAddPageComponent, CoursesPageComponent } from '.';
+import { CourseAddPageComponent, CourseEditPageComponent, CoursesPageComponent } from '.';
 
-export const coursesRoutes: Routes = [
-  { path: AppRoutePath.CourseAdd, component: CourseAddPageComponent },
-  { path: AppRoutePath.Courses, component: CoursesPageComponent },
-];
+export const coursesRoutes: Routes = [ {
+  path: AppRoutePath.CourseAdd,
+  component: CourseAddPageComponent,
+  canActivate: [ AuthGuard ],
+  data: {
+    showBreadcrumbs: true,
+    breadcrumbsSteps: getStaticBreadcrumbsStepsByRoutePath(AppRoutePath.CourseAdd),
```

Comment:
It will be great to use resolver for breadcrumbs getting

Reply:
As far as I know, route resolvers are used to prepare some async data before actual navigation. However, in this case we have all needed data in place before navigation and there is no need to load something asynchronously. That's why route `data` seems for me better option for passing breadcrumbs steps than `resolve`.
    I agree that we could retrieve breadcrumbs steps directly after navigation by current router url and get rid of passing those steps in route `data`. However this approach seems relevant for me only in case of static routes. If we have dynamic route (`/courses/:id`), we won't be able to retrieve breadcrumbs steps by this url in direct way because it can be different each time. We'll need logic for identifying whether route is static or dynamic in this case but it's a kind of redundancy in my opinion.

Reply:
As far as I know, route resolvers are used to prepare some async data before actual navigation. However, in this case we have all needed data in place before navigation and there is no need to load something asynchronously. That's why route `data` seems for me better option for passing breadcrumbs steps than `resolve`.
I agree that we could retrieve breadcrumbs steps directly after navigation by current router url and get rid of passing those steps in route `data`. However this approach seems relevant for me only in case of static routes. If we have dynamic route (`/courses/:id`), we won't be able to retrieve breadcrumbs steps by this url in direct way because it can be different each time. We'll need logic for identifying whether route is static or dynamic in this case but it's a kind of redundancy in my opinion.